metals/src/main/scala/scala/meta/internal/metals/InteractiveSemanticdbs.scala

```diff
@@ -87,18 +85,25 @@ final class InteractiveSemanticdbs(
       val result = textDocumentCache.compute(
         source,
         (path, existingDoc) => {
-          val text = unsavedContents.getOrElse(FileIO.slurp(source, charset))
-          val sha = MD5.compute(text)
-          if (existingDoc == null || existingDoc.md5 != sha) {
-            Try(compile(path, text)) match {
-              case Success(doc) if doc != null =>
-                if (!source.isDependencySource(workspace))
-                  semanticdbIndexer().onChange(source, doc)
-                doc
-              case _ => null
-            }
-          } else
-            existingDoc
+          buffers.get(source).orElse {
```

Comment:
Sorry to bother you, but out of curiosity, what is the difference between the `unsavedContents` (which by the way is still used above in the `shouldTryCalculateInteractiveSemanticdb`, not sure if that's intended now) and `buffers.get(source)`? :thinking:

Reply:
Och, actually `unsavedContents` is unused and it should be used :scream: We use it for to warm up scalafix at the start :thinking: 

I changed it as a first option with fallback to buffers then to filesystem.